src/contexts/NewDelivery.tsx

```diff
@@ -0,0 +1,108 @@
+import useNewDeliverySteps from '@/hooks/useNewDeliverySteps'
+import { NewDeliveryContextValues } from '@/interfaces/newDeliveryContextValues'
+import { createContext, ReactNode, useEffect, useState } from 'react'
+import { SubmitHandler } from 'react-hook-form'
+
+interface NewDeliveryProviderProps {
+  children: ReactNode
+}
+
+export const NewDeliveryContext = createContext<NewDeliveryContextValues>({
+  currentStep: 1,
+  formsData: {},
+  openDialog: false,
+  onCloseDialog: () => {},
+  handleBack: () => {},
+  handleClickOpen: () => {},
+  handleSubmit: () => {},
+  goodsTypeQuantityStep4: [{ pallets: 0, packages: 0, pieces: 0 }],
+  setGoodTypeQuantityStep4: () => {},
+  alertQuantities: '',
+  setAlertQuantities: () => {}
+})
+
+export default function NewDeliveryProvider({ children }: NewDeliveryProviderProps) {
+  const [currentStep, setCurrentStep] = useState(1)
+  const [formsData, setFormsData] = useState<any>({})
+  const [openDialog, setOpenDialog] = useState(false)
+  const [goodsTypeQuantityStep4, setGoodTypeQuantityStep4] = useState([
+    { pallets: 0, packages: 0, pieces: 0 }
+  ])
```

Comment:
lets rather have this one looking something like this.
```ts
interface MoveGood {
  type: string // palette | package | piece
  quantity: number
  zone: string
}
```

This way we know what needs to be sent to the BE. Also you can listen for changes inside this in a `useEffect` and `.map` or `.reduce` through the array to find how much quantity is there for the different types and after that you can find how much is left to be set and put it inside the `alert`.

Maybe `.reduce` will work better where the final result can look like that `{ pallets: 20, packages: 20, pieces: 20 }` this way you know for each type how much quantity you have, and after that you will find how much is left from the quantity set from step 3.

Reply:
Pseudo code
```ts
useEffect(() => {
   const current = step4Items.reduce((acc, current) => {
       if(current.type == 'pallets'){
          acc.pallets += current.quantity
       }
      // same for others
   }, {pallets: 0, packages: 0: pieces: 0})
   
   const itemsLeft = { pallets: step3.pallets - current.pallets, .... //others }

  setAlert(...)
}, [step4Items])
```

Reply:
Fixed